UnityProject/Assets/Scripts/WorldControls.cs

```diff
@@ -161,6 +167,34 @@ public void EstablishConnection()
                 gameStateEventCount++;
             });
     }
+
+    public void NewGameState(string gameStateString)
+    {
```

Comment:
I extracted this out of the EstablishConnection code, we should use this method in there